### PR TITLE
[Enhancement] Support training of BLIP2

### PR DESCRIPTION
**Pull request by fanqiNO1**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

There is no `loss` function for `BLIP2`.

## Modification

1. Add the `loss`function of `BLIP2` based on https://github.com/salesforce/LAVIS/blob/main/lavis/models/blip2_models/blip2_opt.py
2. Fix `init_weights` bug of `Qformer`

## Experiment

Using LoRA, the training result on `COCOCaption` datatset is `BLEU@4`: 42.65, `CIDEr`: 143.84
The result reported in the original paper [BLIP-2: Bootstrapping Language-Image Pre-training with Frozen Image Encoders and Large Language Models](https://arxiv.org/pdf/2301.12597.pdf) is `BLEU@4`: 43.7, `CIDEr`: 145.8
However, I didn't finetune the whole vision backbone.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/blip2/blip2-opt2.7b_8xb32_caption.py [hunk 1]
```diff
@@ -57,7 +57,7 @@
     )
 ]
 
-train_cfg = dict(max_epochs=10)
+train_cfg = dict(by_epocgh=True, max_epochs=10)
```
Comment (fangyixiao18) [inline configs/blip2/blip2-opt2.7b_8xb32_caption.py:60]:
```suggestion
train_cfg = dict(by_epoch=True, max_epochs=10)
```

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by fangyixiao18 — APPROVED** [review]
